### PR TITLE
Issue 4344

### PR DESCRIPTION
**Pull request by Mohamed-Ahmed711**
**Related issue(s):**
* Fixes https://github.com/scandipwa/scandipwa/issues/4344

**Problem:**
* Opacity for selected by default product option is missing on PDP if this product is out of stock

**In this PR:**
* Fixed the Opacity for selected by default product option

### CODE REVIEW

**Comment by aleksandrakorolova** [comment]
1. Opacity is applied for all selected product option. Don't matter if product is out of stock or not - https://drive.google.com/file/d/1ylIr9DvhAzx9piQ6-jGio-NKOmtMm05g/view?usp=sharing
2. Mobile: Red line is displayed above dropdown  
Screenshot 2022-03-07 at 09 38 35

3. Mobile: Can't open dropdown if there is at least one in stock product  - https://drive.google.com/file/d/1XjpNEQ9ClG574bk0WIm77N1n8OxW5Ype/view?usp=sharing

**Comment by aleksandrakorolova** [comment]
4. Opacity - fixed
Error message - still, can reproduce

**Comment by carinadues** [comment]
Decided with Mohamed and Alex that the 2nd part of issue 4 will be fixed separately. Here's the new issue: https://github.com/scandipwa/scandipwa/issues/4445

File: packages/scandipwa/src/component/FieldSelect/FieldSelect.container.js [hunk 1]
```diff
@@ -76,6 +77,19 @@ export class FieldSelectContainer extends PureComponent {
         this.handleIsScrollableList();
     }
 
+    componentDidUpdate() {
+        this.isSelectedOptionAvailable();
+    }
```
Comment (AleksandrsKondratjevs) [inline packages/scandipwa/src/component/FieldSelect/FieldSelect.container.js:82]: Shouldn't there some some kind of condition to not call it each time?

**Review by AleksandrsKondratjevs — CHANGES_REQUESTED** [review]

**Review by AleksandrsKondratjevs — CHANGES_REQUESTED** [review]

**Pull request merged**